gno.land/pkg/integration/testdata/interrealm_mix_call.txtar

```diff
@@ -0,0 +1,206 @@
+gnokey maketx addpkg -pkgdir $WORK/callerrealm -pkgpath gno.land/r/test/callerrealm -gas-fee 1000000ugnot -gas-wanted 20000000 -broadcast -chainid=tendermint_test test1
+
+## execute NonCrossingMutation
+! gnokey maketx call -pkgpath gno.land/r/test/callerrealm -func NonCrossingMutation -gas-fee 1000000ugnot -gas-wanted 10000000 -broadcast -chainid=tendermint_test test2
+stderr 'cannot modify external-realm or non-realm object'
+
+## execute CrossingMutation
+gnokey maketx call -pkgpath gno.land/r/test/callerrealm -func CrossingMutation -gas-fee 1000000ugnot -gas-wanted 10000000 -broadcast -chainid=tendermint_test test2
+
+## execute BorrowedObjectMutation
+gnokey maketx call -pkgpath gno.land/r/test/callerrealm -func BorrowedObjectMutation -gas-fee 1000000ugnot -gas-wanted 10000000 -broadcast -chainid=tendermint_test test2
+
+## execute ObjectMutationRootedAtCaller
+gnokey maketx call -pkgpath gno.land/r/test/callerrealm -func ObjectMutationRootedAtCaller -gas-fee 1000000ugnot -gas-wanted 10000000 -broadcast -chainid=tendermint_test test2
+
+## execute CallMutateObject
+gnokey maketx call -pkgpath gno.land/r/test/callerrealm -func CallMutateObject -gas-fee 1000000ugnot -gas-wanted 10000000 -broadcast -chainid=tendermint_test test2
+
+## execute InspectRealmsCrossing
+gnokey maketx call -pkgpath gno.land/r/test/callerrealm -func InspectRealmsCrossing -gas-fee 1000000ugnot -gas-wanted 10000000 -broadcast -chainid=tendermint_test test2
+
+## execute InspectRealmsNonCrossing
+gnokey maketx call -pkgpath gno.land/r/test/callerrealm -func InspectRealmsNonCrossing -gas-fee 1000000ugnot -gas-wanted 10000000 -broadcast -chainid=tendermint_test test2
+
+## execute MutateObjectWithBorrowCrossing
+gnokey maketx call -pkgpath gno.land/r/test/callerrealm -func MutateObjectWithBorrowCrossing -gas-fee 1000000ugnot -gas-wanted 10000000 -broadcast -chainid=tendermint_test test2
+
+## execute MutateCrossingObjectRootedAtCaller
+! gnokey maketx call -pkgpath gno.land/r/test/callerrealm -func MutateCrossingObjectRootedAtCaller -gas-fee 1000000ugnot -gas-wanted 10000000 -broadcast -chainid=tendermint_test test2
+stderr 'cannot directly modify readonly tainted object'
+
+-- callerrealm/gnomod.toml --
+module = "gno.land/r/test/callerrealm"
+gno = "0.9"
+
+-- callerrealm/main.gno --
+package callerrealm
+
+import (
+    "gno.land/p/test/utils"
+	"gno.land/r/test/borrowrealm"
+)
+
+var obj = borrowrealm.NewObject() // seems to be same thing with obj = borrowrealm.ObjectImpl{}
+
+func NonCrossingMutation(cur realm) {
+    utils.AssertPreviousRealmPkgPath("")
+    utils.AssertCurrentRealmPkgPath("gno.land/r/test/callerrealm")
+
+    borrowrealm.Mutate()
+}
+
+func CrossingMutation(cur realm) {
+    borrowrealm.MutateCrossing(cross)
+}
+
+func BorrowedObjectMutation(cur realm) {
+ 	borrowrealm.Object.Mutate()
+}
```

Comment:
nitpick:
`Borrow` refers to a realm. When an object resides in an external realm, calling a method on this object involves borrowing that external realm. Therefore, I think the function name can be improved to better reflect this concept.

Reply:
Oh thanks I understand what you mean now, I think it's important, I'll change it tomorrow and find something